.buildkite/scripts/steps/cloud/purge_projects.ts

```diff
@@ -0,0 +1,124 @@
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { execSync } from 'child_process';
+import axios from 'axios';
+
+async function getPrProjects() {
+  const match = /^kibana-pr-([0-9]+)-(elasticsearch|security|observability)$/;
+  try {
+    return (
+      await Promise.all([
+        projectRequest.get('/api/v1/serverless/projects/elasticsearch'),
+        projectRequest.get('/api/v1/serverless/projects/security'),
+        projectRequest.get('/api/v1/serverless/projects/observability'),
+      ])
+    )
+      .map((response) => response.data.items)
+      .flat()
+      .filter((project) => project.name.match(match))
+      .map((project) => {
+        const [, prNumber, projectType] = project.name.match(match);
+        return {
+          id: project.id,
+          name: project.name,
+          prNumber,
+          type: projectType,
+        };
+      });
+  } catch (e) {
+    if (e.isAxiosError) {
+      const message = JSON.stringify(e.response.data) || 'unable to fetch projects';
+      throw new Error(message);
+    }
+    throw e;
+  }
+}
+
+async function deleteProject({
+  type,
+  id,
+}: {
+  type: 'elasticsearch' | 'observability' | 'security';
+  id: number;
+}) {
+  try {
+    await projectRequest.delete(`/api/v1/serverless/projects/${type}/${id}`);
+  } catch (e) {
+    if (e.isAxiosError) {
+      const message =
+        JSON.stringify(e.response.data) || `unable to delete ${type} project with id ${id}`;
+      throw new Error(message);
+    }
+    throw e;
+  }
+}
+
+async function purgeProjects() {
+  const prProjects = await getPrProjects();
+  const projectsToPurge = [];
+  for (const project of prProjects) {
+    const NOW = new Date().getTime() / 1000;
+    const DAY_IN_SECONDS = 60 * 60 * 24;
+    const prJson = execSync(
+      `gh pr view '${project.prNumber}' --json state,labels,commits`
+    ).toString();
+    const pullRequest = JSON.parse(prJson);
+    const prOpen = pullRequest.state === 'OPEN';
+    const lastCommit = pullRequest.commits.slice(-1)[0];
+    const lastCommitTimestamp = new Date(lastCommit.committedDate).getTime() / 1000;
+
+    const persistDeployment = Boolean(
+      pullRequest.labels.filter((label: any) => label.name === 'ci:project-persist-deployment')
+        .length
+    );
+    if (prOpen && persistDeployment) {
+      continue;
+    }
+
+    if (!prOpen) {
+      console.log(
+        `Pull Request #${project.prNumber} is no longer open, will delete associated ${project.type} project`
+      );
+      projectsToPurge.push(project);
+    } else if (
+      !Boolean(
+        pullRequest.labels.filter((label: any) =>
+          /^ci:project-deploy-(elasticearch|security|observability)$/.test(label.name)
```

Comment:
There was a small bug here that I resolved by changing the label names from shorthand (oblt -> observability).  I didn't see many references to the shorthand in our codebase and thought it was more consistent with the endpoints.

Also pushed the changes to https://github.com/elastic/kibana/pull/170346/files#diff-316c8c9d17dd2df2124d1f35ed3fcb41059bb542d27b81b3c77e94e6d7ae0860R10